docs/spec.md

```diff
@@ -2,4 +2,253 @@
+```
+
+## 3. Data Sharer transmits a SHLink
+
+The Data Sharer can convey a SHLink by any common means including e-mail, secure messaging, or other text-based communication channels. When presenting a SHLink in person, the Data Sharer can also display the link as a QR code using any standard library to create a QR image from the SHLink URI. 
+
+When sharing a SHLink via QR code, the following recommendations apply:
+
+* Create the QR with Error Correction Level Q
+* Include the [SMART Logo](https://demo.vaxx.link/smart-logo.svg) on a white background over the center of the QR, scaled to occupy 15% of the image area
+
+
+## 4. Data Recipient obtains a SHLink
+
+The Data Recipient can process a SHLink using the following steps.
+
+* Decode the SHLink JSON payload and `POST` to payload's `url`
+    * Method: `POST`
+    * Headers:
+        * `content-type: application/json`
+    * Body: JSON object including
+        * `recipient`: Required. A string describing the recipient (e.g.,the name of an organization or person) suitable for display to the Data Sharer
+        * `passcode`: Conditional. SHALL be populated with a user-supplied Passcode if the `P` flag was present in the SHLink payload
+* Obtain a manifest and follow links to download files
+* Decrypt and process each file
+
+When responding to a manifest request, the Resource Server SHALL reject requests with an invalid Passcode and SHALL enforce a total lifetime count of incorrect Passcodes for a given SHLink, to prevent attackers from performing an exhaustive Passcode search. The error response for an invalid Passcode SHALL use the `401` HTTP status code and the response body SHALL be a JSON payload with
+
+* `remainingAttempts`: number of attempts remaining before the SHL is disabled
+
+### Limitations on Manifest `.files.location` links
+
+The Data Sharer SHALL ensure that `.files.location` links can be dereferenced
+without additional authentication, and that they are short-lived. The lifetime
+of `.files.location` links SHALL NOT exceed one hour. The Data Sharer MAY create
+one-time-use `.files.location` links that are consumed as soon as they are
+dereferenced.
+
+The Data Recipient SHALL treat any manifest file locations as short-lived and
+potentially limited to one-time use. The Data Recipient SHALL NOT attempt to
+dereference a manifest's `.files.location` link more than one hour after
+requesting the manifest, and SHALL be capable of re-fetching the manifest to
+obtain fresh `location` links in the event that they have expired or been
+consumed.
+
+
+The following optional step may occur sometime after a SHLink is shared:
+
+### Optional: Periodically Re-fetch Updated Files 
+When the original QR includes the `L` flag for long-term use, the Data Recipient
+MAY re-fetch contents at any time. When re-fetching data, the Data Recipient
```

Comment:
I think there are some logical limits here. I like re-using the `Expires` header as mentioned below, but "at any time" doesn't give any guidance in the spec to what happens when requests come too quickly.

I assume we should ask servers to specify a maximum request rate on a per-manifest and per-file-per-manifest basis, or perhaps globally per client? Just wondering how to handle poor vs well-behaved clients in the spec.

Reply:
Sure, would love to tighten this down with some implementation experience. For reference though: we haven't done this in SMART on FHIR and face the same "problem" and in practice we've been OK. Servers can always rate limit and in any case servers need to deal with denial of service etc.

Reply:
Fair. I think I didn't fully form my thoughts: an explicit request rate might be interesting, but I'd like the error case of "over some rate" to be well-defined. I'd assume that would mean the caller can expect an HTTP `429`, which is a retryable HTTP code and a client should know that, but it would still be nice to have spelled out.